master-userdata.tpl

```diff
@@ -2,22 +2,27 @@
 set -euo pipefail
 
 # Install Updates and Dependencies from Amazon repositories
-amazon-linux-extras install docker
-yum update -y
-yum -y install java-1.8.0-openjdk make git python3 jq
+sudo amazon-linux-extras install docker
+sudo yum update -y
+sudo yum -y install java-1.8.0-openjdk make git python3 jq
 
 # Configure Docker (Note: Proxies cannot be set by daemon.json with systemd)
-mkdir -p /etc/docker
 
-cat <<EOF >> /etc/sysconfig/docker
+sudo chmod 757 /etc/sysconfig/docker
```

Comment:
I'd prefer not to change permissions of system-managed files, was it not correct during installation?

Reply:
During the installation, I got permission denied on this file. The only way to fix that was by changing the permission and bringing the old permission back after editing the file.